File: Selenium/Activities/Activity17.java
package activities;

import java.util.List;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.Select;

public class Activity17 {

	public static void main(String[] args) {

		WebDriver driver =  new FirefoxDriver();
		
		driver.get("https://v1.training-support.net/selenium/selects");
		
		System.out.println(driver.getTitle());
		
		WebElement dropdown = driver.findElement(By.id("single-select"));

        Select singleSelect = new Select(dropdown);

        singleSelect.selectByVisibleText("Option 2");

        System.out.println("Second option: " + singleSelect.getFirstSelectedOption().getText());

        singleSelect.selectByIndex(3);
       
        System.out.println("Third option: " + singleSelect.getFirstSelectedOption().getText());

        singleSelect.selectByValue("4");
        
        System.out.println("Fourth option: " + singleSelect.getFirstSelectedOption().getText());

        List<WebElement> allOptions = singleSelect.getOptions();
        System.out.println("Options in the dropdown: ");
        for(WebElement option : allOptions) {
            System.out.println(option.getText());
        }

        driver.close();
	}

}

File: Selenium/Activities/Activity17.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.select import Select

with webdriver.Firefox() as driver:

    driver.get("https://v1.training-support.net/selenium/selects")

    print(driver.title)

    single_select = driver.find_element(By.ID, "single-select")
    
    dropdown = Select(single_select)

    dropdown.select_by_visible_text("Option 2")
    
    print("Selected option: ", dropdown.first_selected_option.text)

    dropdown.select_by_index(3)
    print("Selected option: ", dropdown.first_selected_option.text)

    dropdown.select_by_value("4")
    print("Selected option: ", dropdown.first_selected_option.text)

    select_options = dropdown.options
    for option in select_options:
        print("Option: ", option.text)

    driver.close()